### DISCUSSION TITLE
Which loss function is being used in Yolov5?

### DISCUSSION THREAD

**Discussion by lucysumi** [ultralytics/yolov5#9085] [Q&A] [answered]
I am confused which loss function is being used in Yolov5? In loss.py, I saw FocalLoss, QFocalLoss. Is it using Focal loss?

**Comment by glenn-jocher** [comment]
@lucysumi no

BCEloss for obj and cls and CIoU for regression.

**Comment by spacewalk01** [comment, marked as answer]
In loss.py, FocalLoss and QFocalLoss were implemented but were not used in training. For training, BCEWithLogitsLoss is used. If you look at train.py, it uses  ComputeLoss Class which is an implementation of BCEWithLogitsLoss.
https://github.com/ultralytics/yolov5/blob/06831aa9e905e0fa703958f6b3f3db443cf477f3/train.py#L253
ComputeLoss Class:
https://github.com/ultralytics/yolov5/blob/7cef03dddd6fba26fff6748ed1cfdd18208c193e/utils/loss.py#L100

**Reply by lucysumi** [reply]
Thank you very much @spacewalk01
